### ISSUE TITLE
OS Upgrade "Download" button does more than just download new release packages

### ISSUE THREAD

**Issue by rogenry** [pop-os/upgrade#254]
The process that starts after clicking de Download button, in addition to updating the packages of the current release and downloading the packages of the new release, tries to upgrade de following packages: libc-bin libc-dev-bin libc6 libc6:i386 libc6-dev libpop-upgrade-gtk locales pop-upgrade.

I thought it was just download the new release package  and end up with a broken system because the upgrade of the above packages failed. Maybe that upgrade could be done in another step.

**Comment by rogenry** [comment]
>  Just to clarify, you saw pop-upgrade attempt to upgrade itself a second time when you clicked Download?

Yes, after downloading the new release, pop-upgrade attempt to upgrade itself. This is the pop-upgrade messages to the system log:

[pop-upgrade.zip](https://github.com/pop-os/upgrade/files/7873129/pop-upgrade.zip)

**Comment by jacobgkau** [comment]
Ideally, clicking the Download button should display a confirmation dialog box with messaging like this:

```
Pop!_OS __.__ is available!

In order to provide a smooth upgrade experience:

- System-wide (apt) software sources will be reset to Pop!_OS defaults.
- All software packages not installed from a default Pop!_OS source will be removed, and will need to be reinstalled after the upgrade is complete.
- Some user settings, like custom shells and GNOME extensions, will be reset.
- All personal files will be retained, and any software installed from default Pop!_OS sources will remain installed.

If you're ready to proceed, click Download. Otherwise, click Cancel.

(possibly also include the changelog here instead of waiting until after the download's done to show it)
```

Only after clicking Download on this confirmation dialog box should the actual download process begin. Currently, we have no confirmation before we start performing semi-destructive actions in preparation for the upgrade, and we simply attempt to roll back if the user cancels after beginning the download (which has mixed results).

If a user runs `pop-upgrade release upgrade [-f]`, if an upgrade is available, a similar message should be displayed and the user should have a Y/n prompt for whether to proceed. Currently, there's no confirmation at all when the command is run.

@pop-os/ux What do you think about this idea?

**Comment by jacobgkau** [comment]
@maria-komarova It's the OS Upgrade & Recovery page in Settings. It doesn't currently show up in the GUI on 22.04 because we haven't released 24.04 yet. This is what it looks like on 20.04:

<img width="1440" height="900" alt="Image" src="https://github.com/user-attachments/assets/c566ee40-51a5-40fd-8782-eeea4e3ac663" />

Clicking the `Download` button resets the system sources to default, upgrades the current release, removes third-party packages, and downloads the new release.

<img width="1440" height="900" alt="Image" src="https://github.com/user-attachments/assets/444b7e5a-9e5a-49ef-bed6-7b5edddf93b1" />

After the new release has been downloaded, there's another button labeled `Upgrade`:

<img width="1440" height="900" alt="Image" src="https://github.com/user-attachments/assets/10b2f07e-74a1-479b-b1ce-817ab2b4471e" />

Clicking that one displays a confirmation dialog before rebooting to do the offline portion of the upgrade:

<img width="1440" height="900" alt="Image" src="https://github.com/user-attachments/assets/3f3a6d93-02f9-4892-bdff-6eb19cdef5df" />

What I'm suggesting is that a dialog is needed between the first two screenshots. While we try to back the configuration up before performing the "download" actions, it's not always possible to cleanly roll back. And if someone leaves the process in the middle of or after the "download" (before rebooting to install), we attempt to roll back, but it can leave the system in a broken state.

Whether or not this means two dialogs or moving some of the content from the second one into the new earlier one would be up to you.

(Equally important is the need for a yes/no prompt when running `pop-ugprade release upgrade` on the terminal. Currently, it does everything with no confirmation and then prompts to reboot when it's done, so there's nowhere we could possibly explain that we're about to e.g. remove third-party packages.)
